c2rust-bitfields/src/lib.rs

```diff
@@ -28,7 +28,11 @@ pub trait FieldType: Sized {
         let (lhs_bit, rhs_bit) = bit_range;
 
         for (i, bit_index) in (lhs_bit..=rhs_bit).enumerate() {
-            let byte_index = bit_index / 8;
+            let byte_index = if big_endian {
+                field.len() - 1 - (bit_index / 8)
+            } else {
+                bit_index / 8
+            };
```

Comment:
```suggestion
            let byte_index = bit_index / 8;
            let byte_index = if big_endian {
                field.len() - 1 - byte_index
            } else {
                byte_index
            };
```

Reply:
Also, can you refactor all 3 (the 2 below changes, too) of these same computations into some helper function?